### PR TITLE
feat: allow different evaluation aggregations

### PR DESCRIPTION
**Pull request by JoanFM**

### CODE REVIEW

File: jina/optimizers/__init__.py [hunk 1]
```diff
@@ -58,31 +60,44 @@ def __call__(self, response):
         raise NotImplementedError
 
 
-class MeanEvaluationCallback(OptimizerCallback):
+class EvaluationCallback(OptimizerCallback):
     """
-    Calculates the mean of all evaluations during a single :py:class:`FlowRunner`
-    execution from the :py:class:`FlowOptimizer`.
+    Calculates an aggrefation of all evaluations during a single :py:class:`FlowRunner`
```
Comment (florian-hoenicke) [inline jina/optimizers/__init__.py:65]: small typo: aggregation

**Comment by florian-hoenicke** [comment]
@JoanFM I like this change. There was no reason to just have the mean evaluation available. 
Would it require much of a change to get all of them at the same time? Maybe using mean for optimization but saving min, max, median... in the results?

File: jina/optimizers/__init__.py (hunk 1 above)
Comment (JoanFM) [reply jina/optimizers/__init__.py:65]: can u please do a code suggestion?

Comment (JoanFM) [inline jina/optimizers/__init__.py:65]:
```suggestion
    Calculates an aggregation of all evaluations during a single :py:class:`FlowRunner`
```

**Review by florian-hoenicke — DISMISSED** [review]

File: jina/optimizers/__init__.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import os
 from collections import defaultdict
 from typing import Optional
+import numpy as np
```
Comment (bwanglzu) [inline jina/optimizers/__init__.py:4]: move this down to `import yaml`?

File: tests/integration/graph/test_graph_executor.py [hunk 3]
```diff
@@ -60,6 +60,7 @@ def graph():
 
 def test_flow_with_graph_executor(graph, mocker):
     def validate_resp(resp):
+        print(f' type {type(resp)} => {resp}')
```
Comment (bwanglzu) [inline tests/integration/graph/test_graph_executor.py:63]: remove print

**Review by bwanglzu — COMMENTED** [review]
minor things

File: tests/integration/graph/test_graph_executor.py (hunk 3 above)
Comment (JoanFM) [reply tests/integration/graph/test_graph_executor.py:63]: ohh

**Review by bwanglzu — APPROVED** [review]
LGTM!

**Review by Yongxuanzhang — APPROVED** [review]

**Pull request merged**